# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/surface.rs
# Commit: d56eed692ac0
# Full Hash: d56eed692ac0a798b3d38e5ce1ebcdf928a1281e
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-05-30 09:39:43
# Description:
#   Bug 1771547 - Fix task graph dependencies with nested / sibling backdrop-filters r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D147614
# ==============================================================================

diff -r 8e5ff7264e5d -r d56eed692ac0 gfx/wr/webrender/src/surface.rs
--- a/gfx/wr/webrender/src/surface.rs	Sun May 29 23:52:29 2022 +0000
+++ b/gfx/wr/webrender/src/surface.rs	Mon May 30 06:17:30 2022 +0000
@@ -460,6 +460,14 @@
                                             ),
                                         );
 
+                                        // Ensure that the parent task will get scheduled earlier during
+                                        // pass assignment since we are reusing the existing surface,
+                                        // even though it's not technically needed for rendering order.
+                                        rg_builder.add_dependency(
+                                            new_task_id,
+                                            parent_task_id,
+                                        );
+
                                         // Update the surface builder with the now current target for future primitives
                                         tiles.insert(
                                             key,
@@ -518,6 +526,14 @@
                                 ),
                             );
 
+                            // Ensure that the parent task will get scheduled earlier during
+                            // pass assignment since we are reusing the existing surface,
+                            // even though it's not technically needed for rendering order.
+                            rg_builder.add_dependency(
+                                new_task_id,
+                                *parent_task_id,
+                            );
+
                             // Update the surface builder with the now current target for future primitives
                             *parent_task_id = new_task_id;
                         }